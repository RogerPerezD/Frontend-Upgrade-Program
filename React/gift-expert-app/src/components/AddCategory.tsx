import React,{ useState } from 'react';

interface Props{
    setCategories: React.Dispatch<React.SetStateAction<string[]>>;
}

const AddCategory: React.FC<Props> = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleChangeInput = ( event: React.FormEvent<HTMLInputElement>) =>{
        setInputValue( event.currentTarget.value );
    }

    const handleSubmit = async( event: React.FormEvent) =>{
        event.preventDefault();
        console.log('Sumit launch', inputValue);
        if (inputValue.trim().length > 2) {
            setCategories( (categories: string[]) => { 
                return [inputValue,...categories];
            });
        }
        setInputValue('');
    }


    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue } 
                onChange={ handleChangeInput }
            />
            <button type='submit'>Agregar</button>
        </form>
    );
}

export default AddCategory;
