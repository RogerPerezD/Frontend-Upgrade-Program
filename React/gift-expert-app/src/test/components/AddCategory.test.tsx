import { shallow } from "enzyme";
import AddCategory from '../../components/AddCategory';

describe('Test of the component AddCategory', () => {

    const setCategories = () =>{};

    test('should show the component successfully', () => {
        
        const wrapper = shallow( <AddCategory setCategories={setCategories}/>)
    
        expect( wrapper ).toMatchSnapshot();
    });
});