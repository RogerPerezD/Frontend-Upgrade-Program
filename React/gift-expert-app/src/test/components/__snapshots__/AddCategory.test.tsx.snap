// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of the component AddCategory should show the component successfully 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Agregar
  </button>
</form>
`;
